pdr_backend/lake/duckdb_data_store.py

```diff
@@ -349,3 +352,105 @@ def get_query_move_table_data(self, from_table, to_table):
         return (
             f"INSERT INTO {to_table.table_name} SELECT * FROM {from_table.table_name};"
         )
+
+    @enforce_types
+    def export_tables_to_parquet_files(
+        self,
+        seconds_between_exports: int,
+        number_of_files_after_which_re_export_db: int,
+    ):
+        export_folder_path = get_export_folder_path(self.base_path)
+        tables = self.query_data("SHOW TABLES")["name"]
+
+        # Ensure export folder exists
+        os.makedirs(export_folder_path, exist_ok=True)
+
+        for table in tables:
+            table_folder_path = os.path.join(export_folder_path, table)
+            os.makedirs(table_folder_path, exist_ok=True)
+
+            if self._should_nuke_table_folders_and_re_export_db(
+                table_folder_path, number_of_files_after_which_re_export_db, table
+            ):
+                self._nuke_table_folders_and_re_export_db(table_folder_path)
+
+            if not self._should_export(table_folder_path, seconds_between_exports):
+                continue
+
+            self._export_table_to_parquet(table, table_folder_path)
+
+    def _should_export(
+        self, table_folder_path: str, seconds_between_exports: int
+    ) -> bool:
+        max_timestamp_from_parquet = self._get_max_timestamp_from_parquet_files(
+            table_folder_path
+        )
+        current_timestamp = UnixTimeS(int(time.time())).to_milliseconds()
+
+        return (
+            current_timestamp - max_timestamp_from_parquet
+            >= seconds_between_exports * 1000
+        )
+
+    def _get_max_timestamp_from_parquet_files(self, table_folder_path: str) -> int:
+        try:
+            result = duckdb.execute(
+                f"SELECT MAX(timestamp) FROM '{table_folder_path}/*.parquet'"
+            ).fetchone()
+
+            return result[0] if result and result[0] is not None else 0
+        except Exception:
+            return 0  # Assume no data exported yet if there's an error
+
+    def _export_table_to_parquet(self, table: str, table_folder_path: str):
+        max_timestamp_from_db = (
+            self.query_scalar(f"SELECT MAX(timestamp) FROM {table}") or 0
+        )
+        max_timestamp_from_parquet = self._get_max_timestamp_from_parquet_files(
+            table_folder_path
+        )
+
+        if max_timestamp_from_db <= max_timestamp_from_parquet:
+            return  # No new data to export
+
+        parquet_file = f"{table}_{max_timestamp_from_db}.parquet"
+        parquet_file_path = os.path.join(table_folder_path, parquet_file)
+
+        query = f"""
+        COPY (SELECT * FROM {table} WHERE timestamp > {max_timestamp_from_parquet})
+        TO '{parquet_file_path}' (FORMAT 'parquet');
+        """
+        self.execute_sql(query)
+
+    def _should_nuke_table_folders_and_re_export_db(
+        self, table_folder_path: str, file_limit: int, table_name: str
+    ) -> bool:
+        if "bronze" in table_name:
+            return True
+        files = [
```

Comment:
Is there a way to count the files from os, without listing the contents? I think there should be one. Otherwise we are creating a list which we are never using, just to count its number of elements.

Reply:
We could use os.walk() or os.scandir() to go trough the directory and if content is file then update a variable which tracks the number of files

Reply:
https://stackoverflow.com/a/63307131
here is a possible solution that doesn't walk/list the dir